### DISCUSSION TITLE
Dificuldades Odoo Brasil

### DISCUSSION THREAD

**Discussion by molivramento** [OCA/l10n-brazil#2923] [General]
Tem algum tempo que venho acompanhando Odoo, já confirei ele umas 3x, mas sempre desisto, também venho acompanhando outras ferramentas como por exemplo ecommerce Saleor, Frappe e outros como algumas coisas feitas em PHP.

Todos, sem exceção tem exatamente o mesmo problema... BRASIL!
- Não há suporte para nossos métodos de pagamento
- Não há suporte para correios (que ainda bem vem perdendo sua força no monopólio a cada mês)
- Não há suporte a nossos impostos (a tributação nos outros países é muito complexa... comparada a nossa)
- A maioria das empresas brasileiras mal conseguem se manter abertas (não consegue pagar por desenvolvedores ou serviços em nuvem de ponta e etc)

Depois que eu achei OCA, eu fiquei pensando, se tudo é tão trabalhoso, e se esse pessoal praticamente refaz todos os módulos do Odoo, por qual motivo não existe um ERP/Ecommerce/CRM/qualquer outra sigla desenvolvida e mantida por brasileiros, iriamos ganhar na comunidade, as empresas interessadas em colaborar e monetizar isso da forma que bem entenderem, iriamos ter melhor feedback, um projeto mais enxuto e leve (acho que qualquer coisa moderna é mais leve que odoo), a integração de programadores que como eu conseguem fazer algumas coisas, mas não estão no nível de core entre outros.

Isso é apenas a visão de uma pessoa relativamente inocente que quer salvar o mundo!

Pra mim há duas soluções, ou temos algo nosso feito e mantido por brasileiros (hUE-HuE), ou destruímos todos os órgãos públicos e exilamos todos os políticos (até os vereadores) e declaramos todos "persona non grata".

Lógico que apesar de ser um programador, essa minha visão é mais benéfica para usuários que para desenvolvedores, apesar de que eu acho que desenvolvedores ainda assim ganhariam muito mais dinheiro do que hoje com Odoo, visto é IMPOSSÍVEL para um empresa com poucos recursos utilizar Odoo, Saleor, Magento e etc... Todos devem ganhar, e o pequeno deve crescer ou pelo menos melhorar seus recursos.

**Comment by rvalyi** [comment]
Ola @molivramento já teve varias iniciativas de software de gestão open source voltadas pro Brasil. Algumas delas:
- o Stoq
- o FreedomERP
- teve iniciativas com Adempiere e Openbravo...

Alem da OCA teve várias outras tentativas de adaptar o Odoo pro Brasil. Geralmente o que aconteceu foi que essas últimas iniciativas alvejaram empresas pequenas e nao conseguiram bancar o custo da P&D ao longo do tempo. Ja nesse projeto somos algumas empresas com alguns clientes um pouco maior do regime normal que conseguem investir o que um investidor institucional investiria para uma solução de SaaS pequena por exemplo.

Nos accreditamos que cada ano o projeto fica mais maduro e mais facil de usar, até em empresas menores. Pode nao estar no ponto que vc espera hoje, mas a gente segue em frente accteditando que vai chegar lá... 10 anos atrás 90% do trabalho era feito apenas pela minha empresa a Akretion, hoje já tem umas 4 empresas contribuindo forte e a tendência é que vai ter sempre mais...

Um ponto importante é que a gente não quer se divórciar totalmente do Odoo Community na OCA pois accreditamos que quem fizer isso não competiria, seria esmagado até pelo próprio Odoo Enterprise daqui uns anos. Por isso esse esforço gigantesco de migração constante dos módulos ao invez de ter uma versão super completa e simples de usar.

Ao mesmo tempo ninguém impede vc ou alguma outra pessoa de seguir outra iniciativa se accredita que tem mais chances de exito.

Atts.

**Comment by rvalyi** [comment]
mais um comentário importante: vc comentou sobre o fato que tem poucas empresas trabalhando com o Odoo no Brasil. É fondamental vc entender que o mercado do Odoo no Brasil é limitado pelo que se chama de market for lemon ou mercado de abacaxi:
https://saylordotorg.github.io/text_introduction-to-economic-analysis/s19-01-market-for-lemons.html

Pois quem trabalha verdadeiramente com open source, vende apenas serviço e rola esse fenômeno de market for lemon na venda de serviço em volta do Odoo.

Esse problema certamente foi amplificando pela própria Odoo SA que ao invez de regular esse mercado de forma responsável amplifica o problema trocando de integrador de referência a cada 3 ou 4 anos, promovendo direitamente empresas como "gold partner" mesmo tendo 1 case apenas e tendo aberto a empresa o ano anterior, apenas porque essa empresa comprou horas de suporte... Ou deixando empresas da India fraudar a certificação para vc https://twitter.com/cetmix_team/status/1759271042973917284?t=DqDWaP9O6BspvWupAaVirQ&s=19 ...
Ou seja sendo o proprio vendedor de carro ruim que vai lucrar mais a curto prazo da teoria do George Akerlof que ganhou o premio Nobel de economia 2001 com ela.

Nesse mundo selvagem certamente contribuir na OCA é um marcador de qualidade que vai ajudar as empresas que trabalham de verdade com Odoo a se destacar e a melhorar um pouco o cenário caótico que temos hoje com o Odoo, em especial no Brasil.

**Comment by rvalyi** [comment]
Mais um comentário: a gente tem sim que extender quase todos módulos que criam nota fiscal ou tem impostos. MAS mesmo assim e mesmo que é muito trabalhoso, estamos longe de refazer todos módulos do Odoo!! Tem de 3000 a 4000 módulos na OCA e a maioria funciona out of the box com a localização, mesmo que não seja o caso para uns 40% deles que necessitam pequenas extensões como as que criamos neste repositório.

Outro ponto é que depois que essas extensões são maduras e bem testadas, é relativamente facil portar esses módulos para versões mais novas onde beneficiamos de inúmerosas melhorias vindo da Odoo SA e da OCA que a gente nunca conseguiria fazer num projetinho nacional com Stoq ou FreedomERP...

**Comment by augustodinizl** [comment]
Permita-me discordar um pouco de sua "visão", e apresentar alguns contra-argumentos:

> * Não há suporte para nossos métodos de pagamento

Errado, na OCA e em código aberto, temos as integrações: CNAB/brcobranca (bancos em geral), PAGSEGURO e recentemente, testei a versão da odoo s.a para o mercadopago na v.16 , elas FUNCIONAM porém, precisam que a pessoa SAIBA o que está fazendo.

> * Não há suporte para correios (que ainda bem vem perdendo sua força no monopólio a cada mês)

Sabe que a integração via API com os correios é PAGA, né? Se quiser bancar o desenvolvimento e todo o desenvolvimento necessário, pode procurar as principais empresas que desenvolvem na l10n-br.

> * Não há suporte a nossos impostos (a tributação nos outros países é muito complexa... comparada a nossa)

MENTIRA, os impostos são suportados desde a v9 @rvalyi pode me corrigir se eu estiver errado.

> * A maioria das empresas brasileiras mal conseguem se manter abertas (não consegue pagar por desenvolvedores ou serviços em nuvem de ponta e etc)

HAHAHAHAHAHAHAHAHAHAHAHAHA, cara, tá de brincadeira,né?

> e se esse pessoal praticamente refaz todos os módulos do Odoo, por qual motivo não existe um ERP/Ecommerce/CRM/qualquer outra sigla desenvolvida e mantida por brasileiros,

Não é "refazer", é ADAPTAR, EXPANDIR, ou seja: LOCALIZAR.

Sobre o resto, nem vou perder tempo com esse blá blá blá...

> visto é IMPOSSÍVEL para um empresa com poucos recursos utilizar Odoo

eu montei uma instância da V16 para produção gastando:
R$ 1400 em um mini pc dell optiplex.
R$ 40 em um domínio no registro.br.
R$ 9,99 em um e-mail catchall com a godaddy.com
R$ 0,00 com a conectividade do meu server para a internet (usando cloudflared)
ou seja: gastando menos que uma TELEVISÃO, eu montei uma instancia para produção.

Man, na boa, gasta MAIS TEMPO ESTUDANDO/FAZENDO, e menos tempo RECLAMANDO.

#PAZ, e se precisar de alguma orientação, ajuda ou coisa parecida, pode perguntar, mas não abusa.

**Reply by molivramento** [reply]
Me desculpe senhor deus da verdade, até o teu comentário eu estava entendendo muito bem o que realmente se passou no mercado brasileiro (obrigado @rvalyi ) mas quando chegou no teu, deu até nojinho de ler, mas já que vc passou tanto redigindo, não poderia fazer essa desfeita.

Agora explicando para todos que tomaram como OFENSA (e eu nem me esforcei pra isso... foi pura inocência).

Passei vários meses estudando documentação Oddoo, e perambulando pela comunidade, em todo lugar que eu passei era gente falando da estrutura do oddoo, falando mal da empresa que não se importa com nada além de lucro, e bom, os problemas brasileiros de fisco nem vou entrar no mérito, TODO ANO ou MENOS, tira importo, bota imposto, muda imposto, correios com seu maravilhoso preço e XML, nem vou falar dos métodos de pagamentos.

vc me chamou de mentiroso, como se eu tivesse falado algo sabendo e mesmo assim falado outra coisa por prazer e não desconhecimento (ainda tenho dúvida se realmente é tão maravilhoso como senhor deus da verdade falou), eu poderia deixar a quinta séria fluir e xingar vc e sua mamãe, mas vamos evitar isso, somos adultos neh (pelo menos fisicamente eu sou).

De vc não quero nem água no deserto, quem dirá orientação/ajuda/ou coisa parecida.

Meu ERP (django + django-ninja + async + quasar framework):
R$ 42 VPS contabo
R$ 3.33 dominio .com.br
R$ 13.33 dominio .com
R$ 0 provedor de email no VPS

Parece que o meu é maior que o teu...

**Comment by marcos-mendez** [comment]
Eu ri 🤣

Em qui., 22 de fev. de 2024 às 17:45, Augusto D. Lisbôa <
***@***.***> escreveu:

**Comment by antoniospneto** [comment]
@molivramento 

Penso que sua visão sobre o Odoo, a comunidade OCA e a localização pode estar um pouco equivocada. Não sei o que o levou a pensar que tudo precisa ser refeito, pois esse não é o objetivo da equipe aqui. Todos os módulos da versão community da Odoo SA e os módulos internacionais da OCA são aproveitados no projeto de localização brasileira. Um dos pontos mais fortes do framework Odoo é a modularidade; não se trata de um módulo substituir outro, mas sim de módulos que estendem os já existentes, complementando-se mutuamente.

De fato, o Brasil apresenta muitos desafios, com suas numerosas regras, legislações e exigências fiscais. No entanto, sem dúvida, seria muito mais difícil sem o framework do Odoo. Por que abandonar as vantagens do Odoo e começar tudo do zero? Não faz sentido.

**Reply by molivramento** [reply]
Eu vi muita briga e descontentamento tanto com o oddoo (empresa) e produto também, não fiz critica aos devs brasileiros que fazem parte do OCA, só achei a discussão poderia valer a pena nem que fosse para eu de fato entender, eu nem levei em consideração que os devs tem sua renda ou parte dela daqui.

Achei que sei lá, com tantos problemas seria mais viável um projeto brasileiro, julguei que teríamos e talvez menos descontentamentos.

Mas outra coisa que não levei em consideração, enquanto eu fazia o ERP pra loja aqui, eu percebi que ela tinha algumas coisas meio que apenas dela, e ERPs como oddoo e SAP deixam meio que o caminho livre pra personalização.

Eu realmente não me vejo trabalhando em ERP para abraçar todo e qualquer tipo de seguimento isso deve ser insano.

Por exemplo, aqui nos utilizamos REDE, GetNet e Pix (Nu Bank, Inter, Pic Pay) para pagamento, não utilizamos correios, trabalhamos nos 3 seguimentos macro (produção, varejo e serviço) e outras coisas pontuais. Mesmo com tudo isso foi relativamente fácil fazer sob medida. Mas seria muito doloroso fazer com que tudo isso fosse extensível para outros comércios/industrias, com outros fluxos.

Novamente peço desculpas, não foi por arrogância foi realmente sincero e sem a intenção de ofender. (exceto a resposta anterior, talvez eu tenha sim tentado ofender, apenas um retribuição...)

**Comment by Bmessiazy** [comment]
Eu venho falando desde 2021 que esse @rvalvy eh um atraso para o Brasil. O cara nem fala português direito? E tira onda de entender o que fala. 
Tínhamos que fazer uma moção com o maximo de assinaturas de quem usa Odoo no Brasil pra tirar ele da localização OCA.
Nunca vai funcionar, porque ele eh LIMITADO no universo paralelo de ZÉ ROELA QUE ELE CRIOU.
CONTÉM COMIGO PRA ASSINAR ESSA DOCUMENTO.
PARABÉNS PELA CORAGEM! 
Abraço 

Enviado do Yahoo Mail para iPhone


Em segunda-feira, abril 22, 2024, 11:28 AM, Marcelo Livramento ***@***.***> escreveu:




Eu vi muita briga e descontentamento tanto com o oddoo (empresa) e produto também, não fiz critica aos devs brasileiros que fazem parte do OCA, só achei a discussão poderia valer a pena nem que fosse para eu de fato entender, eu nem levei em consideração que os devs tem sua renda ou parte dela daqui.

Achei que sei lá, com tantos problemas seria mais viável um projeto brasileiro, julguei que teríamos e talvez menos descontentamentos.

Mas outra coisa que não levei em consideração, enquanto eu fazia o ERP pra loja aqui, eu percebi que ela tinha algumas coisas meio que apenas dela, e ERPs como oddoo e SAP deixam meio que o caminho livre pra personalização.

Eu realmente não me vejo trabalhando em ERP para abraçar todo e qualquer tipo de seguimento isso deve ser insano.

Por exemplo, aqui nos utilizamos REDE, GetNet e Pix (Nu Bank, Inter, Pic Pay) para pagamento, não utilizamos correios, trabalhamos nos 3 seguimentos macro (produção, varejo e serviço) e outras coisas pontuais. Mesmo com tudo isso foi relativamente fácil fazer sob medida. Mas seria muito doloroso fazer com que tudo isso fosse extensível para outros comércios/industrias, com outros fluxos.

Novamente peço desculpas, não foi por arrogância foi realmente sincero e sem a intenção de ofender. (exceto a resposta anterior, talvez eu tenha sim tentado ofender, apenas um retribuição...)

**Comment by marcelsavegnago** [comment]
@Bmessiazy a resposta do @molivramento foi para o @augustodinizl  e não para o @rvalyi .

Não seja troll :D heheheheheheh aqui ta todo mundo na paz....

@molivramento acho que é pertinente sim a discussão e pelo menos da minha parte não vejo nada de ofensivo.. Porém, não vejo que o caminho seja abraçar o mundo.. Hoje trabalhamos com base no Odoo e sem colocar a mão em tudo dele já é bem trabalhoso.

**Comment by marcelsavegnago** [comment]
@Bmessiazy, se você realmente deseja fazer uma contribuição valiosa para a comunidade, encorajo você a participar ativamente. Contribuir com uma revisão de PR ou se aprofundar em uma issue significativa pode ser muito mais benéfico do que se envolver em discussões desnecessárias. Na verdade, é mais simples do que parece: basta começar. Não há necessidade de sentir medo ou vergonha. Ficaria muito contente em ver uma PR sua, mesmo que seja um commit simples. Sua participação é bem-vinda!

**Comment by molivramento** [comment]
Olha, vejo que mexi num vespeiro, mas eu realmente cai de paraquedas no meio do turbilhão, acho melhor fechar essa discussão, daqui a pouco vai rolar até divorcio... motivo: Oddoo.

Vamos todos ser amigos outra vez, vcs estão bem com oddoo, eu estou bem com meu app direcionado a demanda especifica, tá todo mundo feliz e ganhando dinheiro!

**Comment by marcos-mendez** [comment]
Amem! 🙌

Em seg., 22 de abr. de 2024 às 13:49, Marcelo Livramento <
***@***.***> escreveu:

**Comment by Bmessiazy** [comment]
Não trabalho com o Zé Roela do Raphael! Ja passei da idade de aceitar um ditadorzinho francês temperamental que dá chiliques todos os momentos.. enche a boca que tem mais de 10 anos de OcA.. e não tem uma versão confiável nem da 14, nem da 15 nem da 16? Fala pra mim onde roda uma versão OCA LOCALIZAÇÃO SEM REMENDOS ??
Já vai sair a 18 e a situação do BR eh essa, se fosse seria essa OCA já tinha tirado ele de lá mas no fundo todos esses caras da OCA usam como vitrine pra fechar negócios paralelos, tirando o Mileo que eu sei que eh Sério!! 
Ele faz a OCA  br ser refem dele para interesses próprios, duvido que sem ela teria vendas. O Brasil. 
Simples assim!! O que esperar se um cara que lê seu PR entende, aprende e altera umas coisinhas para se apropriar da ideia ?
Mau caráter SIMPLES ASSIM!! 
Essa eh a VERDADE ACREDITA QUEM QUISER! 


Enviado do Yahoo Mail para iPhone


Em segunda-feira, abril 22, 2024, 12:55 PM, Marcel Savegnago ***@***.***> escreveu:




@Bmessiazy eu sugiro caso vc de fato queria melhorar algo na comunidade, basta contribuir, seria muito mais útil revisar uma PR ou ler uma issue decente do que ter que lidar com discussões por conta de ego ou sei lá oq..... mas veja, está fácil na verdade, bota a mão na massa, não precisa ter medo ou vergonha caso seja o caso. Eu sinceramente adoria ver uma PR sua nem que seja com um unico commit bobo.

**Comment by marcos-mendez** [comment]
Alguem muta esse cara pelo amor de deus!

Em seg., 22 de abr. de 2024 às 15:25, Bmessiazy ***@***.***>
escreveu:
